app/service/FigureService.scala

```diff
@@ -113,16 +115,35 @@ class FigureService(figPath: String, figMobilePath: String) {
         throw new EntityNotFoundException("Unable to find abstract with uuid = " + abstr.uuid)
 
 
-      val file = new File(figPath, fig.uuid)
+      // In a docker environment '/target/universal/stage' is used when resolving '.'
+      // in a relative path. 'java.io.File' seems to properly resolve the relative paths
+      // even in a docker environment, but the third party library scrimage cannot work
+      // with these file descriptors and requires file descriptors created with absolute paths.
+      //   As a workaround the file not found exception is caught and the docker container
+      // root path is used to create file descriptior with absolut paths appropriate for
+      // the docker environment.
+      var figure_file = new File(figPath, fig.uuid)
+      var mobile_file = new File(figMobilePath, fig.uuid)
+
+      try {
+        var mobile_image = Image.fromFile(figure_file)
+      } catch {
+        case nofile: NoSuchFileException => {
+          val figure_path = Paths.get("/srv", "gca", figPath, fig.uuid).normalize.toString
+          val mobile_path = Paths.get("/srv", "gca", figMobilePath, fig.uuid).normalize.toString
```

Comment:
Would a more general solution be always convert paths to abs paths everywhere (or maybe directly from the settings)?  Is there a reason that wouldn't work?